backend/de.metas.contracts/src/main/java/de/metas/contracts/bpartner/process/C_BPartner_MoveToAnotherOrg.java

```diff
@@ -0,0 +1,161 @@
+import de.metas.process.ProcessPreconditionsResolution;
+import de.metas.product.ProductId;
+import lombok.NonNull;
+import org.compiere.SpringContextHolder;
+import org.compiere.model.I_C_BPartner;
+
+import javax.annotation.Nullable;
+import java.time.LocalDate;
+
+public class C_BPartner_MoveToAnotherOrg extends JavaProcess implements IProcessPrecondition,
+		IProcessParametersCallout,
+		IProcessDefaultParametersProvider
+{
+	@Param(parameterName = "AD_Org_Target_ID", mandatory = true)
+	private OrgId p_orgTargetId;
+
+	@Param(parameterName = "M_Product_ID")
+	private ProductId p_membershipProductId;
+
+	@Param(parameterName = "Date_OrgChange", mandatory = true)
+	private LocalDate p_startDate;
+
+	@Param(parameterName = "IsShowMembershipParameter", mandatory = true)
+	private boolean isShowMembershipParameter;
+
+	final OrgChangeRepository orgChangeRepo = SpringContextHolder.instance.getBean(OrgChangeRepository.class);
+	final OrgChangeService service = SpringContextHolder.instance.getBean(OrgChangeService.class);
+
+	@Override
+	protected String doIt() throws Exception
+	{
+		final I_C_BPartner bpartnerRecord = getProcessInfo().getRecord(I_C_BPartner.class);
+
+		final OrgChangeRequest orgChangeRequest = OrgChangeRequest.builder()
+				.bpartnerId(BPartnerId.ofRepoId(bpartnerRecord.getC_BPartner_ID()))
+				.startDate(p_startDate)
+				.membershipProductId(p_membershipProductId)
+				.orgFromId(OrgId.ofRepoId(bpartnerRecord.getAD_Org_ID()))
+				.orgToId(p_orgTargetId)
+				.build();
+
+		service.moveToNewOrg(orgChangeRequest);
+
+		return MSG_OK;
+	}
+
+	@Override
+	public ProcessPreconditionsResolution checkPreconditionsApplicable(final @NonNull IProcessPreconditionsContext context)
+	{
+		if (!context.isSingleSelection())
+		{
+			return ProcessPreconditionsResolution.rejectBecauseNotSingleSelection();
+		}
+
+		return ProcessPreconditionsResolution.accept();
+	}
+
+	@Override
+	public void onParameterChanged(final String parameterName)
+	{
+		if ("AD_Org_Target_ID".equals(parameterName) || "Date_OrgChange".equals(parameterName))
+		{
+			if(p_orgTargetId == null)
+			{
+				return;
+			}
+			final BPartnerId partnerId = BPartnerId.ofRepoId(getRecord_ID());
+
+			final LocalDate orgChangeDate = CoalesceUtil.coalesce(p_startDate, SystemTime.asLocalDate());
+
+			final OrgChangeBPartnerComposite orgChangePartnerComposite = orgChangeRepo.getByIdAndOrgChangeDate(partnerId, orgChangeDate);
+
+			if ((orgChangePartnerComposite.hasMembershipSubscriptions())
+					&& service.hasAnyMembershipProduct(p_orgTargetId))
+			{
+				isShowMembershipParameter = true;
+			}
+			else
+			{
+				isShowMembershipParameter = false;
+			}
+		}
+	}
+
+	@Nullable
+	@Override
+	public Object getParameterDefaultValue(final IProcessDefaultParameter parameter)
+	{
+		if ("Date_OrgChange".equals(parameter.getColumnName()))
+		{
+			return SystemTime.asLocalDate().plusDays(1);
+		}
+
+		// if("M_Product_ID".equals(parameter.getColumnName()))
```

Comment:
is this a missed TODO? if not, pls remove the commented code or write why it's still needed